### PR TITLE
Course Automation: Checking link validity in the course github text descriptions

### PR DESCRIPTION
**Pull request by Chen-Zidi**
# Course automation: Checking link validity in the course github text descriptions

## Members

Chen, Zidi: zidi@kth.se, https://github.com/Chen-Zidi
Chen, Sihan sihanc@kth.se, https://github.com/Spycsh

## Proposal

We want to check the validity of the deadlinks in the course github 
text descriptions, for example, readme files. Because sometimes 
the links might be not accessible or out of date.

## Solution
TODO

### CODE REVIEW

**Comment by cesarsotovalero** [comment]
Hi @Chen-Zidi,

What your action will do when finding a broken link?

**Comment by Spycsh** [comment]
Hi @cesarsotovalero,
after finding a broken link, our idea is:  
1. to create a github action and whenever there is a pull request, the action should automatically parse the children or descendant pages under the course website (https://github.com/KTH/devops-course) to find all the broken links
2. to show the detail information (time of when the action find the broken link, the corresponding commit log and who writes this dead link, using git blame maybe) of all the broken links in a list of one issue

We are still thinking whether the trigger event of above action is a pull request or a timeout, and whether we need to parse all the children pages or just the pages produced in the new pull request. Of course we should do some filtering of the valid files to parse (emphasis on README file, txt file).

**Comment by cesarsotovalero** [comment]
Hi @Spycsh,

> whether we need to parse all the children pages or just the pages produced in the new pull request

Checking only the links introduced in the PR is enough.
Then the ideal would be closing the PR with a message explaining that one of the links is broken. The other option is opening an issue with explaining the problem (not ideal because probable the person who opened the PR will is not going to take action).

File: contributions/course-automation/zidi-sihan/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+## Proposal
+
+We want to check the validity of the deadlinks in the course github 
+text descriptions, for example, readme files. Because sometimes 
+the links might be not accessible or out of date.
+
+
+## Solution
+TODO
```
Comment (cesarsotovalero) [inline contributions/course-automation/zidi-sihan/README.md]: Please update with the details of your proposed solution.

**Review by cesarsotovalero — CHANGES_REQUESTED** [review]

**Review by cesarsotovalero — APPROVED** [review]

**Comment by cesarsotovalero** [comment]
@Chen-Zidi LGTM now.

**Pull request merged**